commands/deals_daemon_test.go

```diff
@@ -0,0 +1,91 @@
+package commands_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/filecoin-project/go-filecoin/fixtures"
+	th "github.com/filecoin-project/go-filecoin/testhelpers"
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+)
+
+func TestListDeals(t *testing.T) {
+	tf.IntegrationTest(t)
+
+	clientDaemon := th.NewDaemon(t,
+		th.KeyFile(fixtures.KeyFilePaths()[1]),
+		th.DefaultAddress(fixtures.TestAddresses[1]),
+	).Start()
+	defer clientDaemon.ShutdownSuccess()
+
+	minerDaemon := th.NewDaemon(t,
+		th.WithMiner(fixtures.TestMiners[0]),
+		th.KeyFile(fixtures.KeyFilePaths()[0]),
+		th.DefaultAddress(fixtures.TestAddresses[0]),
+		th.AutoSealInterval("1"),
+	).Start()
+	defer minerDaemon.ShutdownSuccess()
+
+	minerDaemon.RunSuccess("mining", "start")
+	minerDaemon.UpdatePeerID()
+
+	minerDaemon.ConnectSuccess(clientDaemon)
+
+	addAskCid := minerDaemon.MinerSetPrice(fixtures.TestMiners[0], fixtures.TestAddresses[0], "20", "10")
+	clientDaemon.WaitForMessageRequireSuccess(addAskCid)
+	dataCid := clientDaemon.RunWithStdin(strings.NewReader("HODLHODLHODL"), "client", "import").ReadStdoutTrimNewlines()
+
+	proposeDealOutput := clientDaemon.RunSuccess("client", "propose-storage-deal", fixtures.TestMiners[0], dataCid, "0", "5").ReadStdoutTrimNewlines()
+
+	splitOnSpace := strings.Split(proposeDealOutput, " ")
+	dealCid := splitOnSpace[len(splitOnSpace)-1]
+
+	expectedClientOutput := fmt.Sprintf(`{
+	"minerAddress": "%s",
+	"pieceCid": {
+		"/": "QmbHmUVAkqZjQXgifDady7m5cYprX1fgtGaTYxUBBTX3At"
+	},
+	"proposalCid": {
+		"/": "%s"
+	},
+	"state": 2
+}`, fixtures.TestMiners[0], dealCid)
+
+	expectedMinerOutput := fmt.Sprintf(`{
+	"minerAddress": "%s",
+	"pieceCid": {
+		"/": "QmbHmUVAkqZjQXgifDady7m5cYprX1fgtGaTYxUBBTX3At"
+	},
+	"proposalCid": {
+		"/": "%s"
+	},
+	"state": 7
+}`, fixtures.TestMiners[0], dealCid)
+
+	t.Run("with no filters", func(t *testing.T) {
+		listClientDealsOutput := clientDaemon.RunSuccess("deals", "list").ReadStdoutTrimNewlines()
+		assert.Equal(t, expectedClientOutput, listClientDealsOutput)
+
+		listMinerDealsOutput := minerDaemon.RunSuccess("deals", "list").ReadStdoutTrimNewlines()
```

Comment:
These tests are difficult to parse (although they are correct). We also have no examples that demonstrate that the filters can filter some deals while leaving others.

It would be better if the setup were to have daemon A make a deal with daemon B, and then have daemon B make a deal with daemon B. Then run all the tests against just daemon A.